code/headers/display_adapter.hpp

```diff
@@ -1,8 +1,21 @@
 #pragma once
 
 #include <hwlib.hpp>
-
+#include <display_cursor.hpp>
 namespace r2d2::display {
+    /*
+    * Class display_c is the base class for all displays in R2D2. It inherits
+    * from hwlib::window.
+    * 
+    * @tparam Cursor_Count is the amount of cursors to store. Most of the time 
+    * this is equal to the number claimed cursors in the display_cursor enum.
+    * 
+    * @tparam Display_Size_Width is an uint8_t representing the width of the display
+    * 
+    * @tparam Display_Size_Height is an uint8_t representing the height of the display
+    * 
+    */
+    template<std::size_t Cursor_Count, uint8_t Display_Size_Width, uint8_t Display_Size_Height>
```

Comment:
Why is it a std::size_t. A size_t should work as well?

Reply:
Yes, a size_t would work just as fine. But it is not defined in the standard which one I should use. Seeing as how they both function exactly the same I don't see why you see this as a fault needing correction. On the other side, std::size_t is defined in the C++ standard and thus make for more code conformity.

Reply:
Well the 5 other occurrences in the file don't use it.

Reply:
Fixed in c2dfd0d4e42b03c5f9eeb9234d5f7f92fd062ad3

Reply:
I rather have the size_t than the std::size_t btw

Reply:
I can't find the base of your opinion in the styleguide.

Reply:
I prefer `std::size_t`, it's scope is narrower.
Also when given a choice between a _c solution_ and a _c++ solution_ in a _c++ project_, all else being equal i prefer the _c++ solution_.